lib/TPP/Dialect/VNNI/TransformOps/VNNITransformOps.cpp

```diff
@@ -0,0 +1,92 @@
+//===- VNNITransformOps.cpp - Implementation of VNNI transform ops--====//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "TPP/Dialect/VNNI/TransformOps/VNNITransformOps.h"
+#include "TPP/Dialect/Tpp/TppOps.h"
+#include "TPP/Dialect/VNNI/VNNIOps.h"
+#include "TPP/Transforms.h"
+#include "mlir/AsmParser/AsmParser.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Dialect/Transform/IR/TransformDialect.h"
+#include "mlir/Dialect/Transform/IR/TransformInterfaces.h"
+#include "mlir/Interfaces/ViewLikeInterface.h"
+#include "mlir/Parser/Parser.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "llvm/ADT/TypeSwitch.h"
+#include "llvm/Support/Casting.h"
+
+using namespace mlir;
+using namespace mlir::transform;
+using namespace mlir::tpp;
+//===----------------------------------------------------------------------===//
+// Transform op registration
+//===----------------------------------------------------------------------===//
+
+namespace {
+
+class VNNITransformDialectExtension
+    : public transform::TransformDialectExtension<
+          VNNITransformDialectExtension> {
+public:
+  using Base::Base;
+
+  void init() {
+    registerTransformOps<
+#define GET_OP_LIST
+#include "TPP/Dialect/VNNI/TransformOps/VNNITransformOps.cpp.inc"
+        >();
+  }
+};
+} // namespace
+
+namespace {
+// A simple pattern rewriter that implements no special logic.
+class SimpleRewriter : public PatternRewriter {
```

Comment:
this should not be needed anymore after my llvm bump. Simply include the file: https://github.com/llvm/llvm-project/commit/a8850312c106d71f9e35fd902f9dcd3c4ac0a690. Bonus point if you do that also for other transform we have.

Reply:
I have fixed this transform in  the other patch https://github.com/plaidml/tpp-mlir/pull/189. I will update this patch with trivialpatternrewriter for the other transform.